storm-core/src/jvm/org/apache/storm/executor/spout/SpoutExecutor.java

```diff
@@ -79,8 +79,11 @@ public SpoutExecutor(final Map workerData, final List<Long> executorId, Map<Stri
         this.spoutThrottlingMetrics = new SpoutThrottlingMetrics();
     }
 
-    @Override
     public void init(final Map<Integer, Task> idToTask) {
+        while (!stormActive.get()) {
```

Comment:
I just follow previous executor implementation, and invoking init() to multiple times is not intended not only for while statement, but also rest of statements.